### PR TITLE
docs: add docs for new features

### PR DESCRIPTION
**Pull request by straker**
Closes issue: #2754

### CODE REVIEW

**Comment by straker** [comment]
I'm not sure why it thinks I'm merging in 4 commits when those commits are already in develop?

File: doc/API.md [hunk 1]
```diff
@@ -793,6 +796,28 @@ axe.cleanup(resolve, reject);
 
 `resolve` takes no arguments and `reject` takes a single argument that must be a string or have a toString() method in its prototype.
 
+### API Name: axe.setup
+
+Setup axe-cores internal VirtualNode tree and other required properties required to run functions in `axe.commons`.
```
Comment (stephenmathieson) [inline doc/API.md:801]:
```suggestion
Setup axe-cores internal `VirtualNode` tree and other required properties required to run functions in `axe.commons`.
```

File: doc/API.md [hunk 2]
```diff
@@ -793,6 +796,28 @@ axe.cleanup(resolve, reject);
...
+
+The signature is:
+
+```js
+axe.setup(DomNode);
+```
+
+`DomNode` - is an optional DomNode to use as the root of the VirtualNode tree. Default is `document.documentElement`.
```
Comment (stephenmathieson) [inline doc/API.md:809]:
```suggestion
`DomNode` - is an optional DOM node to use as the root of the `VirtualNode` tree. Default is `document.documentElement`.
```

File: doc/API.md [hunk 3]
```diff
@@ -793,6 +796,28 @@ axe.cleanup(resolve, reject);
...
+
+### API Name: axe.teardown
+
+Cleanup the VirtualNode tree and internal caches. `axe.run` will call this function at the end of the run so there's no need to call it yourself afterwards.
```
Comment (stephenmathieson) [inline doc/API.md:813]:
```suggestion
Cleanup the `VirtualNode` tree and internal caches. `axe.run` will call this function at the end of the run so there's no need to call it yourself afterwards.
```

File: doc/API.md [hunk 4]
```diff
@@ -822,6 +847,26 @@ axe.utils.querySelectorAll(virtualNode, 'a[href]');
 
 An Array of filtered HTML nodes.
 
+#### API Name: axe.utils.getRule
+
+##### Description
+
+Get an axe-core Rule instance by ID.
```
Comment (stephenmathieson) [inline doc/API.md:854]:
```suggestion
Get an axe-core `Rule` instance by ID.
```

File: doc/API.md [hunk 5]
```diff
@@ -822,6 +847,26 @@ axe.utils.querySelectorAll(virtualNode, 'a[href]');
...
+axe.utils.getRule('color-contrast');
+```
+
+##### Parameters
+
+- `ruleId` - The ID of the rule.
+
+##### Returns
+
+An axe-core Rule instance.
```
Comment (stephenmathieson) [inline doc/API.md:868]:
```suggestion
An axe-core `Rule` instance.
```

**Review by stephenmathieson — APPROVED** [review]
Minor suggestions to improve readability. LGTM tho 👍

**Review by stephenmathieson — APPROVED** [review]

**Pull request merged**